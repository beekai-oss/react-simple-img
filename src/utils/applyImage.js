// @flow

function applyStyle(target, endStyle, withoutPlaceholder) {
  if (withoutPlaceholder) {
    target.style.opacity = 1; // eslint-disable-line
  } else {
    const style = target.getAttribute('style');
    target.setAttribute('style', `${style && style.includes('opacity') ? '' : 'opacity: 0;'}${style}${endStyle}`);
  }
}

export default function applyImage(target: any, image: Image, src: string) {
  /* eslint-disable */
  target.src = src;
  if (target.dataset.srcset) {
    target.srcset = target.dataset.srcset;
  }

  target.style.visibility = 'visible';

  const withoutPlaceholder = target.getAttribute('data-placeholder') === 'false';
  /* eslint-enable */
  const currentTarget = withoutPlaceholder ? target : target.nextSibling;

  if (currentTarget || withoutPlaceholder) {
    target.addEventListener('load', () => {
      if (target) {
        const endStyle = target.getAttribute('data-end-style') || '';
        applyStyle(currentTarget, endStyle, withoutPlaceholder);
        target.removeEventListener('load', applyStyle);
      }
    });
  }
  window.__REACT_SIMPLE_IMG__.imgLoadingRefs.delete(target);
}
