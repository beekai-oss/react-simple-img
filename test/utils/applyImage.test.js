import applyImage from '../../src/logic/applyImage';

describe('applyImage', () => {
  it('should update mount images state when applyImage is called', () => {
    const setAttributeSpy = jest.fn();
    const getAttributeSpy = jest.fn();
    const addEventListener = jest.fn();
    const deleteSpy = jest.fn();
    window.__REACT_SIMPLE_IMG__ = {
      observer: {
        unobserve: () => {},
      },
      imgLoadingRefs: {
        set: () => {},
        delete: deleteSpy,
      },
    };
    expect(
      applyImage(
        {
          addEventListener,
          src: 'test',
          dataset: {
            srcset: 'srcset',
          },
          srcset: 'srcset',
          style: {
            visibility: 'hidden',
          },
          getAttribute: getAttributeSpy,
          nextSibling: {
            setAttribute: setAttributeSpy,
          },
        },
        'loadedImg',
      ),
    ).toMatchSnapshot();
    expect(getAttributeSpy).toHaveBeenCalled();
    expect(addEventListener).toHaveBeenCalled();
    expect(deleteSpy).toHaveBeenCalled();

    window.__REACT_SIMPLE_IMG__ = undefined;
  });
});
