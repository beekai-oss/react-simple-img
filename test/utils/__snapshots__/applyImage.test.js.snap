// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applyImage should update mount images state when applyImage is called 1`] = `undefined`;
